src/zenml/utils/artifact_utils.py

```diff
@@ -418,3 +420,43 @@ def upload_artifact(
         )
 
     return response.id
+
+
+def get_producer_step_of_artifact(
+    artifact: "ArtifactResponseModel",
+) -> "StepRunResponseModel":
+    """Get the step run that produced a given artifact.
+
+    Args:
+        artifact: The artifact.
+
+    Returns:
+        The step run that produced the artifact.
+
+    Raises:
+        RuntimeError: If the run that created the artifact no longer exists.
+    """
+    if not artifact.producer_step_run_id:
+        raise RuntimeError(
+            f"The run that produced the artifact with id '{artifact.id}' no "
+            "longer exists. This can happen if the run was deleted."
+        )
+    return Client().get_run_step(artifact.producer_step_run_id)
+
+
+def get_artifacts_of_pipeline_run(
+    pipeline_run: "PipelineRunResponseModel",
+) -> List["ArtifactResponseModel"]:
+    """Get all artifacts produced during a pipeline run.
+
+    Args:
+        pipeline_run: The pipeline run.
+
+    Returns:
+        A list of all artifacts produced during the pipeline run.
+    """
+    artifacts: List["ArtifactResponseModel"] = []
+    for step in pipeline_run.steps.values():
+        if step.status == ExecutionStatus.COMPLETED:  # no cached artifacts
```

Comment:
Is this something we want to parameterize, e.g. allow users to also fetch artifacts that were cached? I guess in most cases when I'd look for artifacts/outputs of a run, I wouldn't actually care if it's cached/new and rather have a complete list?

Reply:
Makes sense. Maybe I'll add a separate property for each 👍

Reply:
Added `run.produced_artifacts` as separate property.

Reply:
Also added `step.parent_steps` as property, as you suggested on #1648